### DISCUSSION TITLE
How can i get the @Root that i need in @FieldResolver

### DISCUSSION THREAD

**Discussion by ashalfarhan** [MichalLytek/type-graphql#859] [Q&A]
Any suggest how can i get reference to the `User` type or models that i want to get the data from

Here's my code:

`UserResolver.ts`
<img width="472" alt="Screen Shot 2021-04-17 at 02 03 26" src="https://user-images.githubusercontent.com/70800415/115072407-c4d59180-9f21-11eb-8547-1d45e1f1794a.png">

`User` in entity / typedef
<img width="271" alt="Screen Shot 2021-04-17 at 02 03 42" src="https://user-images.githubusercontent.com/70800415/115072578-f9e1e400-9f21-11eb-91d8-980e86e6bd47.png">

`UserModel`
<img width="279" alt="Screen Shot 2021-04-17 at 02 03 59" src="https://user-images.githubusercontent.com/70800415/115072992-1a11a300-9f22-11eb-9fed-6e5d570f57a0.png">

Then here is what i got the error
<img width="1272" alt="Screen Shot 2021-04-17 at 02 04 48" src="https://user-images.githubusercontent.com/70800415/115073199-301f6380-9f22-11eb-9bf2-92fbf828f74f.png">

I have no idea how to solve this.
Btw, i'm just using mongoose, not typegoose

But when i got someone code has this:
```typescript
import { ObjectId } from "mongodb";

export type Ref<T> = T | ObjectId;
```
It still gives me the error
Hope anyone can help me 🙏🏼 

Thankyou!

**Comment by MichalLytek** [comment]
Show the resolver where you return user document

**Comment by ashalfarhan** [comment]
This one?

<img width="468" alt="Screen Shot 2021-04-17 at 09 42 19" src="https://user-images.githubusercontent.com/70800415/115099613-4698df80-9f61-11eb-9d12-def1b215ba94.png">

**Reply by MichalLytek** [reply]
As described in typegoose related issue and showed in the examples, you need to call `.toObject()` in order to transform mongoose document into a JS object.

**Reply by ashalfarhan** [reply]
Yes, i thought its difficult to setup type-graphql and typegoose, but yes, when i add the typegoose it made easier, thank you!

**Reply by MichalLytek** [reply]
Typegoose is just decorators instead of `new Schema`. There's still mongoose under the hood, with `.populate()` etc.
